nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp

```diff
@@ -73,8 +83,22 @@ inline BT::NodeStatus GoalUpdater::tick()
         last_goal_received_time.seconds(), goal_time.seconds());
     }
   }
-
   setOutput("output_goal", goal);
+
+  if (last_goals_received_.header.stamp != rclcpp::Time(0) && !last_goals_received_.poses.empty()) {
+    auto last_goals_received_time = rclcpp::Time(last_goals_received_.header.stamp);
```

Comment:
What timestamp-checking logic would you have here? In this implementation I'm just checking the top-level one. We could also check all the individual goals timestamps and reject the update if at least one is outdated

Reply:
I think that we should check if any are newer. That would indicate that a new message has come in that was not previously known.

I read this check as looking at the topic's goal and the input port's goal and selecting the topic's if its newer (and a topic update is present). So, I think the analog here would be to check the goals list's header.

I know in the `geometry_msgs/PoseStampedArray` PR we discussed removing it because we couldn't think of a good reason to include it -- I think _this_ is the good reason to include it. You could have that message type defining a set of goals so the goals themselves have their own stamps as dynamic constraints, yet the global header is referring to the time the goals themselves were computed. 

Sorry to flip flop :(

Reply:
I did not get what you are proposing to compare exactly? Is it if any of the topic PoseStamped are more recent than any of the input PoseStamped? Or if any of the topic PoseStamped are more recent than the stamp of input PoseStampedArray?

If we are keeping the top-level header with the meaning "the time the goals themselves were computed", then I don't see why we would not just keep it as is (compare current and incoming top-level stamps).

Reply:
I'm saying that I think we should do the same comparison as the singular goal case; Check the global header only to make sure its a newer computed goal than the previous one. Don't look at the vector of PoseStamped stamps. I think we're on the same page :-)

Reply:
> I would rather not do that in this PR.

I agree. We should do that after the actual `geometry_msgs` option is merged so we don't fliplash people twice.

Reply:
So to be clear, this PR is blocked both by the `geometry_msgs` option AND by a PR that switches all other users of `std::vector<geometry_msgs::msg::PoseStamped>` to `PoseStampedArray`.

Reply:
OK - we could merge this now with `nav2_msgs` version and update geometry_msgs once that's merged - and then do the rest of the switches to use it. But also happy to wait if that's what you prefer